src/bespokelabs/curator/request_processor/base_request_processor.py

```diff
@@ -168,7 +220,13 @@ async def acreate_request_file(
                 # Get the generic request from the map function
                 request = prompt_formatter.create_generic_request(dataset_row, dataset_row_idx)
                 await f.write(json.dumps(request.model_dump(), default=str) + "\n")
-        logger.info(f"Wrote {end_idx - start_idx} requests to {request_file}.")
+
+        num_written = end_idx - start_idx
+        metadata_dict = {"num_jobs": num_written}
```

Comment:
nit: num_requests 

also... instead of `metadata_i.json` we can also just store a single `metadata.json` that maps from request file name to metadata object. Just an option to consider. 

Then we can also store another field which is the total number of request files (which also helps with validation) E.g. if in the middle of batch creation the worker dies and only some of the request files are present we wouldn't catch that if we just looked at each file.

Reply:
@RyanMarten thanks for the comment - having single metadata json would require some changes to the structure (since currently these async functions don't really return anything).

I changed it to still having `metadata_i.json`, but instead of only verifying if all exist it also explicitly checks each file separately, and regenerating only missing ones. This takes care of the second case you mention as well.